### PR TITLE
Hide unused fields for Guardian on Rule Edit page.

### PR DESCRIPTION
**Pull request by mozbhearsum**
We're in the process of adding support for the new Guardian product to Balrog. One of the things that will be different about its updates, is that the update URL will not contain most of the fields that Firefox updates do. In order to avoid confusion, I'd like to hide these fields to avoid any confusion.

At this time, the backend is not going to be smart enough to refuse to accept Rules for Guardian that contain these fields, so I'm specifically _not_ hiding them on the Rules view. On the off chance they get set by accident, it would be confusing to have values but not show them.

### CODE REVIEW

**Comment by helfi92** [comment]
I am back from PTO :) I will review this as soon as I can.

**Comment by helfi92** [comment]
@mozbhearsum How can I test this? I don't seem to see "Guardian" in the list of rules.

**Comment by mozbhearsum** [comment]
Ah, you'll need to create a rule for Guardian. It can be basically whatever you want (point at a Firefox Release or whatever - doesn't matter). Let me know if you have any trouble with that.

**Comment by mozbhearsum** [comment]
> When I try to create a new rule with product "Guardian" , I am seeing an error panel with message update_type cannot be set to null/empty when scheduling insertion of a new rule. Any idea why this is happening?

Hm. I'm unable to reproduce this - it always gets set to `minor` for me. What exactly is your STR?

> Would it try to set an OS version even though it's a restricted property?

Great question. I will double check, and make sure we don't pass it to the backend.

**Comment by helfi92** [comment]
> Hm. I'm unable to reproduce this - it always gets set to minor for me. What exactly is your STR?

Works now after running `docker-compose build` :)

**Comment by helfi92** [comment]
> Great question. I will double check, and make sure we don't pass it to the backend.

Sounds good!

**Comment by mozbhearsum** [comment]
I actually managed to repro the `update_type` issue as well - which was caused by `update_type` being considered an unsupported field. It is, technically (we ignore it on the backend for Guardian), but it's required by the database schema, so for now we can't totally ignore it :(. We can improve this later IMO.

**Comment by mozbhearsum** [comment]
> > Would it try to set an OS version even though it's a restricted property?
> 
> Ah, it turns out we are still sending the other properties. I added a console statement in `handleCreateRule` and was able to see
> 
> Screen Shot 2019-10-10 at 4 55 30 PM
> 
> The osVersion is unsupported for the Guardian product but was set. We should probably fix that.

Are you sure? The call to `filterProductData` happens within the call to `updateSC`, so if you're just logging `data` you'd see stale contents. I tested these scenarios and wasn't able to get `osVersion` set - but I may have done something wrong!

**Comment by mozbhearsum** [comment]
(see https://github.com/mozilla-frontend-infra/balrog-ui/pull/226/files#diff-b6d56a3781defa9b6101ab1eaa826383R242)

**Comment by helfi92** [comment]
Oh, sorry about that. I had the log statement above filterProductData. I'll check again.

File: src/views/Rules/Rule/index.jsx [hunk 1]
```diff
@@ -181,6 +182,13 @@ function Rule({ isNewRule, user, ...props }) {
     }
   };
 
+  const productSupportsField = field =>
+    !(
+      rule.product in RULE_PRODUCT_UNSUPPORTED_PROPERTIES &&
+      RULE_PRODUCT_UNSUPPORTED_PROPERTIES[rule.product].includes(field)
+    );
+  const filterProductData = data =>
+    pick(Object.keys(data).filter(field => productSupportsField(field)), data);
```
Comment (helfi92) [inline src/views/Rules/Rule/index.jsx]: `.filter(field => productSupportsField(field))` -> `.filter(productSupportsField)`

**Pull request merged**